### PR TITLE
feat: enabled setTimeout/clearTimeout APIs

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
Added setTimeout and clearTimeout that enables delayed code execution.

Fixes # (issue)

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Manual
- Cypress
- Jest

### Test plan
- [ ] https://github.com/appsmithorg/TestSmith/issues/2074
- [ ] https://github.com/appsmithorg/TestSmith/issues/2076
- [ ] https://github.com/appsmithorg/TestSmith/issues/2077

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/timeout_spec.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        shouldCreateNewJSObj: true,
+        prettify: true,
+      },
+    );
+    agHelper.Sleep(2000);
+    jsEditor.RunJSObj();
+    agHelper.Sleep(3000);
+    agHelper.AssertContains("resolved");
+  });
+  it.only("verifies code execution order when using setTimeout", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/timeout_spec.ts:87]: Do you want to run only this case from this spec?
And will the set timeouts of one it function not hinder with another during run of this spec?

Comment (arunvjn) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/timeout_spec.ts]: Accidental commit. Reverted

**Review by ramsaptami — APPROVED** [review]

**Review by ohansFavour — DISMISSED** [review]

**Review by ramsaptami — APPROVED** [review]

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/timeout_spec.ts [hunk 2]
```diff
@@ -0,0 +1,218 @@
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
+const jsEditor = ObjectsRegistry.JSEditor;
+const agHelper = ObjectsRegistry.AggregateHelper;
+const locators = ObjectsRegistry.CommonLocators;
+const apiPage = ObjectsRegistry.ApiPage;
+const deployMode = ObjectsRegistry.DeployMode;
+
+describe("Tests setTimeout API", function() {
+  it("Executes showAlert after 3 seconds and uses default value", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/timeout_spec.ts:9]: Please index the 'it' functions, eg; `it("1. Executes`

Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/timeout_spec.ts:9]: Spec name can be CamelCase - SetTimeout_Spec.ts.

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks good.

**Pull request merged**